### PR TITLE
Add `wrap_func_args` parameter

### PR DESCRIPTION
**Pull request by joseprio**
Adds an output parameter to wrap function definitions, which may speed up JS parsing

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "homepage": "https://github.com/terser-js/terser",
   "author": "Mihai Bazon <mihai.bazon@gmail.com> (http://lisperator.net/)",
   "license": "BSD-2-Clause",
-  "version": "4.1.4",
+  "version": "4.1.5",
```
Comment (L2jLiga) [inline package.json]: I pretty sure this should not be changed

File: README.md [hunk 2]
```diff
@@ -1036,6 +1037,10 @@ can pass additional arguments that control the code output:
   function expressions. See
   [#640](https://github.com/mishoo/UglifyJS2/issues/640) for more details.
 
+- `wrap_func_args` (default `false`) -- pass `true` to wrap function expressions
+  that are passed as arguments. See
+  [OptimizeJS]https://github.com/nolanlawson/optimize-js) for more details.
```
Comment (L2jLiga) [inline README.md]: You missed parenthesis😉 
`[OptimizeJS](https://github.com/nolanlawson/optimize-js)`

File: package.json (hunk 1 above)
Comment (fabiosantoscode) [reply package.json]: Yeah, I handle version bumps :)

File: lib/output.js [hunk 3]
```diff
@@ -186,6 +186,7 @@ function OutputStream(options) {
         webkit           : false,
         width            : 80,
         wrap_iife        : false,
+        wrap_func_args   : false,
```
Comment (fabiosantoscode) [inline lib/output.js]: Please go ahead and make this true by default :) some things in Terser don't provide the least possible byte size and that's OK.

File: test/compress/issue-427.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    beautify = {
+        wrap_func_args: true,
+    }
+    input: {
+        console.log(function() {
+            return "test";
+        });
+    }
+    expect_exact: 'console.log((function(){return"test"}));'
+}
```
Comment (fabiosantoscode) [inline test/compress/issue-427.js]: Would be cool to add a test which disables negate_iife, and enables `wrap_iife` and `wrap_func_args` by default.

The test would ensure that `console.log((function() { return 42 })())` doesn't add an extra pair of parens.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## v4.1.5
+
+ - Added a new output parameter, `wrap_func_args`, which wraps function arguments in parenthesis to speed up parsing in some browsers.
+
```
Comment (fabiosantoscode) [inline CHANGELOG.md]: I also deal with the changelog, no worries here. You can delete these changes :)

**Comment by fabiosantoscode** [comment]
Don't be disheartened @joseprio! We only need these few changes and will not bother you with other details :)

**Comment by joseprio** [comment]
@fabiosantoscode I bumped the version because it was mentioned in `CONTRIBUTING.md`, `Pull Request Process`, but happy to revert those changes!

**Comment by joseprio** [comment]
Other than removing the version bump and adding the missing parenthesis in `README.md`, I added an extra test for when the `wrap_func_args` parameter is disabled.

**Comment by fabiosantoscode** [comment]
Thanks for letting me know @joseprio, I've updated CONTRIBUTING.md to better reflect our real process, and to remove the duplicate code of conduct included within.

Merging!

**Pull request merged**